google/cloud/internal/async_read_write_stream_tracing.h

```diff
@@ -112,14 +113,19 @@ class AsyncStreamingReadWriteRpcTracing
  private:
   Status End(Status status) {
     if (!context_) return status;
```

Comment:
hm, it looks like we want to say: `if (!span_)` now. Both branches reset the `span_` ptr, but only one resets the `context_` ptr.

I think we would have a new crash if:
- `Start()` returns false
- `Finish()` is called
- the destructor runs.

Reply:
Done. Good catch! Thanks!